service/src/main/java/eu/europa/ec/fisheries/uvms/activity/service/mapper/FishingActivityMapper.java

```diff
@@ -288,6 +289,48 @@ protected Instant getCalculatedStartTime(FishingActivity fishingActivity) {
         return null;
     }
 
+    protected Instant getCalculatedEndTime(FishingActivity fishingActivity) {
+        if (fishingActivity == null) {
+            return null;
+        }
+        DateTimeType occurrenceDateTime = fishingActivity.getOccurrenceDateTime();
```

Comment:
I think the code is a little bit easier to understand if you refrain from getting this variable until just before you use it at the end of the method